### DISCUSSION TITLE
ClimateHunt Hackathon

### DISCUSSION THREAD

**Discussion by petermr** [petermr/petermr#48] [Announcements]
# Discussion specific to NIPGR event 2022-09-24

(* will copy over some material from earlier )

# Structure of discussion

There will be specific **COMMENT**s within this thread. Everyone can **REPLY**, but comments and replies can only be edited by the author. **PLEASE DO NOT CREATE NEW COMMENTS**, as the titles are selected to be searchable. 

It's likely we may move to a different format to finalise the material for delegates. 

The comment topics include:

## purpose of hackathon and takeaways
* understanding and creating climate knowledge (esp. IPCC)
* understanding and developing human/machine knowledge systems
* making new contacts
* feedback knowledge and technology to `climateHunt`

## likely attendees
* GY invitees
 - GY Invitees: Dr. P Maini, Scientific Secretary, Office of PSA to Govt of India
 - The Climate Collective (x3)
 - Mr. Vimal Kumar, Finovista
 - Head, Delhi City Knowledge & Innovation Cluster (DRIIV)
 - Vertiver Agency
 - Office of S&T, Swedish Embassy of India,
 - IORA Ecologicals
 - Mr. Rahul Kulshreshtha, Strategic Alliance Division, Office of PSA to GoI

* NIPGR staff and students
* openVirus members 
* openVirus alumni
* openVirus collaborators
    - openClimate (Shannon Dosemagen)
    - openClimate (TiB, Simon Worthington)
    - SDG Accelerator (UniGeneva)
    - Wikimedia (Daniel Mietchen, ?Anasuya Sengupta)

### general introduction
* GY and invited attendees

### hackathon introduction
* zoom. practice (GY)
* interns and alumni present, explain, demo
    - programme and purpose
    - hackathon technology
### delegate introductions
* communal 'video-on' and wave
### timetable from GY
```
INAUGURAL SESSION: 10:30AM to 11:30 AM BST
10:30: Welcome Address & Intro to Delhi Cluster (Dr. G Yadav NIPGR)
10:35: Introduction to Semantic Climate (Dr. Peter Murray-Rust, Cambridge)
10:40: Inaugural Address Dr. P Maini (Sc Secy, O/o PSA to GoI)
10:50: UNESCO/SDG Accelerator Award Announcement for our Team! (PMR/SH)
10:53: Keynote Address by Ashish Chaturvedi (Head, UNDP India)
11:03: Open Citizen Science (Simon Worthington, LIBER Europe)
11:10: WikiMedia for Academics (Daniel Mietchen, WikiData)
11:15: Vote of Thanks (By GY)
11:20: Coffee/Comfort Break (10 minutes)
MAIN SESSION: 11:30AM to 2 PM BST
11:30: Aim of The Hackathon (Ms Shweata Hegde, Developer of Climate Tools)
11:45: Thematic Grouping and Assignment of IPCC Chapters
12:00: Shared Learning across 4-5 Climate Themed Break Out Rooms
13:30: Presentations by Rapporteurs of Each Group
13:45: Discussion on Future Activities & Closure (Facilitated by GY and PMR)
```

## online booklet ("flyer")
The first of several online and possibly real-life hackathons.
"Climate Knowledge Hunt" is a half-day hackathon bringing together a wide variety of people and materials.
### aims
* to explore the IPCC reports and make them more accessible
*  to share our experience, skills, tools and knowledge
* to see if we can create new materials/thoughts/tools, etc.
* to form new collaborations
* to see how modern tools (hands-on where possible) can help make sense of complex information

If we have done even some of these things, the hackathon will be a success

### approach

We are learning as we go! Some of our tools are tested, but others are new.

### event infrastructure
* hosted from ?NIPGR?
* Zoom
* Breakout rooms
* Github 
    - should all delegates have Github IDs?
* Google Colab (can everyone use it?)
* Communal message tool:

### pre-registration
* check 1 week in advance that all can use it.
* pre-installation of tools?
* pre-installtation of content?
##

**Comment by petermr** [comment]
# Purpose of Hackathon

Topics:
* understanding and creating climate knowledge (esp. IPCC)
* understanding and developing human/machine knowledge systems
* making new contacts
* feedback knowledge and technology to climateHunt

If you want to add material REPLY to this COMMENT. You (and only you)  can edit that REPLY

**Comment by petermr** [comment]
# Attendees

This is for information on attendees

**Comment by petermr** [comment]
# Programme

**Comment by petermr** [comment]
# Content

**Reply by petermr** [reply]
... telling a story to the delegates. This is probably the most important thing in the whole event. The delegates will want to know:
* WHY are we doing this (whether it's the whole Hackathon or a single cell in Jupyter)
* WHAT are we going to do? (before you or delegates press any buttons they should understand what is going to happen)
* HOW are we going to do it?
* WHAT do we expect to happen?
Then they can press the button
* WHAT actually happened?

You will know more than the delegates about the software and they will often know more than us about climate change. So think of yourselves as equals in a conversation. Even if you only know a little bit about why / what / how it's more than nothing.

Then you need the story of semanticClimate and why it's an important project. Explain it in simple terms. Here's a story we can expand on:
* The United Nations IPCC is the most authoritative resource in the world on climate change. 
* It releases its findings reports every 5 years on everything climate related. 2022  (AR6) was the most recent and runs to over 10,000 pages of text, tables, diagrams and references.
* It's written in English for specialists and uses a great deal of jargon (technical terms, abbreviations, acronyms). These are usually explained only once. There's a PDF glossary.
* It's published as large PDF documents (sometimes 3000 pages). There's a Table of Contents but no index. Many chapters reference each other but links have to be traversed by humans.

How do we make it easier to read and use? Everything here is automatic!

* We make it semantic (machine-understandable) by
  - splitting into sections/subsections/lists/sentences etc. (This is non-trivial)
  - turning it into HTML
  - indexing all sections/sentences with unique IDs so we can link to anywhere in the 10000 pages
* we make specialised dictionaries of all the words and phrases in the report.
* we add meaning to these with Wikipedia/Wikidata and translations to many human languages
* we annotate the HTML with hyperlinks to other sections and the dictionaries.
* we create an index so readers can search precisely for technical terms.
* we extract words from the images 

The semantic reports can also be input into other software (e.g. layered onto maps, read into data  analysis tools).

**Comment by petermr** [comment]
# Organization

**Reply by petermr** [reply]
# breakout rooms

For 40 attendees , maybe 6 breakout rooms with 5-10 people in each

**Reply by petermr** [reply]
# Google Colab

What permissions  are needed to use Google Colab?

**Reply by ShweataNHegde** [reply]
People would have to make a copy of the Colab Notebook to their drive to run them. I don't think we need to give any permissions.

**Comment by petermr** [comment]
# Tutorials

**Reply by ShweataNHegde** [reply]
## Google Colab 101
We might want to have an introductory Colab Notebook to explain how to run them (with screenshots, etc) 

## Core Notebook
- The Notebook would be the same for all delegates (same chapters, dictionaries, etc)
- We should
     -  define constants at the beginning (check: https://github.com/petermr/climate/blob/master/ipcc_analysis.ipynb)
     -  add more textual material (but ensure that the delegates run all cells between texts
     -  use comments within the code to indicate the order in which the cells should run

**Comment by petermr** [comment]
# Tasks TODO
(MoSCoW - Must/Should/Could/Wont)

# Critical (MUST)

## Event "booklet"
### Components
What you may encounter in this event. With **brief** narrative

## Core notebook

## Chapters in structured HTML
### annotations

## "Scripts" for helpers

## 

# Highly desirable (SHOULD)

# Useful (COULD)

#

**Comment by petermr** [comment]
# Overview (Flyer/Booklet) for the Hackathon 2022-09-24

A "1-page" "booklet" for potential delegates **before** the event

## what's our Hackathon?

A  communal fun semi-structured halfday exploration of the IPCC reports and how ro find information in them.

## what can I expect?

## what can I bring?
* bring your questions and/or knowledge of climate.
* IT experience (e.g. creating webpages) is valuable but NOT required
* willingness and excitement to work in small teams

## what experience and skills do I need?
No special skills other than using a browser. The core software is mounted in the cloud.

TO BE CONTINUED

**Comment by petermr** [comment]
# Activities and demos:

## breakout rooms
organized by
* Chapter
 currently:
  - Chapter02 (emissions) Abhishek
  - Chapter07 (Agriculture) Kavita and Ambreen
  - Chapter08 (Cities and other settlements) Priti
  
 Each chapter has
  - semantic HTML text
  - a dictionary linked to Wikidata
  
 
* experience
  - introductory (hackathon and Zoom)
  - commandline techniques
  - dictionaries

**Reply by ShweataNHegde** [reply]
### 2022-09-17 
-  Lightening talk by each chapter champion to help delegates decide which chapter to go with
- An admin to create and manage breakout rooms
#### Demonstration in the first half of the hackathon
- A Colab Notebook for each chapter; same recipe different input
#### Activity in the second half of the hackathon
- Manual classification: Given an automatically generated word cloud, can the delegate identify which chapter it came from? 
- Finding important sections of reports

**Comment by priti-chahal** [comment]
# BREAKOUT ROOM DEBRIEF
## Title: Urban system and other settlements (Chapter 08)
## Instructor: Priti
**Duration of hackathon session**: 1 hour
**Recording**: Was started in the mid-session, but not sure if the session was fully recorded. Possibility of recording the session fully is there. 
**Moving between chatrooms**: Yes, possible. But the way is breakout room A – main room – breakout room B – main room – breakout room C (A= chapter2, B=chapter7, C=chapter8).
**Did organizer's come into the rooms**? Yes.
**Have screenshots of the google colab**: Did not create chapter screenshots, as the link for google colab came at the end, but 2 minutes summary of the chapter was delivered by me via screenshots on ppt before beginning the chapter exploration could have been very useful. 

## Problems during meeting: 

1. I was not able to share my screen (due to technical issues) so Ananya take over the screen. I was handling chat box at that time.
2. Initially, the google colab link was shared in the main breakout room session. Thus, no one was able to use it as a link. The link had to be shared again on the breakout room chats for them to become usable.
3. Ananya had restarted the session after 5 min. Everyone was curious about what the next would be.
4. During the initial phase some were confused because they were opening the link in repository and unable to find the right way to open google colab.
5. During the initial installation, some were not able to find the results of first cell then we tried to solve that by giving the actual path to the files.
6. In the first cell, one person did not run the “restart runtime” button after the 1st run, and that led to further problems in the downstream cells.
7. During PDF to HTML conversion, almost everyone wanted to know why we were not completing the running of the cell. Ananya explained that a lot of temporary files were formed currently.  So, we were not running them to full and that we had already a html of the chapter ready. 
8. One of the issues was someone tried to run the next cell before running the first one so there was an error message.
9. The delegates wanted to create dictionary related to their topic of interest from IPCC report.
10. One of the delegates was not able to get the result after running the cell but by refreshing the files he got the result.
11. Some wanted to do code level stuff of how to create the word cloud as well as other sections.
12. At the end, 1 hour appeared to be a limited time some of them was not able to get results by their own.

## Suggestions: 

1. A five-minute chapter-wise instruction in the breakout room only based on screenshots would have primed everyone for the session.
2. The instructions must begin with asking everyone to share any problems in the chat immediately, and also to speak out if the speed of the instructor is fast or slow.
3. The delegates took about 15 min to get used to the colab and running of the cells. But after that, just clicking of the cell to play and get the result bored them. Then, they wanted to get to applying these codes on new data and see what they could extract out of it. So, for the future hackathons, after half the instructions have been done, and the delegates have got the hang of what to do, there may be descriptions of how the code is run and performed on IDE and then brought back to colab as a play button code, to maintain the instructions more engaging and maintaining a difficulty level for the whole session, while also showing to the delegates how what we are doing is cutting edge as well as difficult to achieve.
4. PDF to HTML is the core functionality of our tools and allows other tool functions to be explored even on the new data. So, if this can be done without glitches, exploring of new documents as well as comparing of the chapters would be possible.
5. There has to a set of instructions for the breakout room chapter instructor to follow. There has to room for spontaneity, but there must also be to-do lists that allows for proper breakdown of what to do when, and how to do it.
6. A central co-ordinator who is not handling any other issue would be very useful, as he/she would be responsible only to check out for the issues occurring in the real time and alerting the right person who could be involved in solving it.

**Comment by petermr** [comment]
This is a good report, Priti!
Can you add it to the discussions as Markdown, please.


On Fri, Sep 30, 2022 at 7:36 AM priti-chahal ***@***.***>
wrote:

> BREAKOUT ROOM DEBRIEF Title: Urban system and other settlements (Chapter
> 08) Instructor: Priti
>
> *Duration of hackathon session*: 1 hour
> *Recording*: Was started in the mid-session, but not sure if the session
> was fully recorded. Possibility of recording the session fully is there.
> *Moving between chatrooms*: Yes, possible. But the way is breakout room A
> – main room – breakout room B – main room – breakout room C (A= chapter2,
> B=chapter7, C=chapter8).
> *Did organizer's come into the rooms*? Yes.
> *Have screenshots of the google colab*: Did not create chapter
> screenshots, as the link for google colab came at the end, but 2 minutes
> summary of the chapter was delivered by me via screenshots on ppt before
> beginning the chapter exploration could have been very useful.
> Problems during meeting:
>
>    1. I was not able to share my screen (due to technical issues) so
>    Ananya take over the screen. I was handling chat box at that time.
>    2. Initially, the google colab link was shared in the main breakout
>    room session. Thus, no one was able to use it as a link. The link had to be
>    shared again on the breakout room chats for them to become usable.
>    3. Ananya had restarted the session after 5 min. Everyone was curious
>    about what the next would be.
>    4. During the initial phase some were confused because they were
>    opening the link in repository and unable to find the right way to open
>    google colab.
>    5. During the initial installation, some were not able to find the
>    results of first cell then we tried to solve that by giving the actual path
>    to the files.
>    6. In the first cell, one person did not run the “restart runtime”
>    button after the 1st run, and that led to further problems in the
>    downstream cells.
>    7. During PDF to HTML conversion, almost everyone wanted to know why
>    we were not completing the running of the cell. Ananya explained that a lot
>    of temporary files were formed currently. So, we were not running them to
>    full and that we had already a html of the chapter ready.
>    8. One of the issues was someone tried to run the next cell before
>    running the first one so there was an error message.
>    9. The delegates wanted to create dictionary related to their topic of
>    interest from IPCC report.
>    10. One of the delegates was not able to get the result after running
>    the cell but by refreshing the files he got the result.
>    11. Some wanted to do code level stuff of how to create the word cloud
>    as well as other sections.
>    12. At the end, 1 hour appeared to be a limited time some of them was
>    not able to get results by their own.
>
> Suggestions:
>
>    1. A five-minute chapter-wise instruction in the breakout room only
>    based on screenshots would have primed everyone for the session.
>    2. The instructions must begin with asking everyone to share any
>    problems in the chat immediately, and also to speak out if the speed of the
>    instructor is fast or slow.
>    3. The delegates took about 15 min to get used to the colab and
>    running of the cells. But after that, just clicking of the cell to play and
>    get the result bored them. Then, they wanted to get to applying these codes
>    on new data and see what they could extract out of it. So, for the future
>    hackathons, after half the instructions have been done, and the delegates
>    have got the hang of what to do, there may be descriptions of how the code
>    is run and performed on IDE and then brought back to colab as a play button
>    code, to maintain the instructions more engaging and maintaining a
>    difficulty level for the whole session, while also showing to the delegates
>    how what we are doing is cutting edge as well as difficult to achieve.
>    4. PDF to HTML is the core functionality of our tools and allows other
>    tool functions to be explored even on the new data. So, if this can be done
>    without glitches, exploring of new documents as well as comparing of the
>    chapters would be possible.
>    5. There has to a set of instructions for the breakout room chapter
>    instructor to follow. There has to room for spontaneity, but there must
>    also be to-do lists that allows for proper breakdown of what to do when,
>    and how to do it.
>    6. A central co-ordinator who is not handling any other issue would be
>    very useful, as he/she would be responsible only to check out for the
>    issues occurring in the real time and alerting the right person who could
>    be involved in solving it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/petermr/petermr/discussions/48#discussioncomment-3769268>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFTCS3RT3JW2BKPMFSVTV3WA2C7HANCNFSM56RYZG7A>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Peter Murray-Rust
Founder ContentMine.org
and
Reader Emeritus in Molecular Informatics
Dept. Of Chemistry, University of Cambridge, CB2 1EW, UK
